modules/ROOT/pages/traversal-framework/index.adoc

```diff
@@ -0,0 +1,67 @@
+:description: The Neo4j Traversal Framework Java API.
+
+:org-neo4j-graphdb-Direction-both: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Direction.html#BOTH
+
+
+[[traversal]]
+= The Traversal Framework
+
+[abstract]
+--
+This section provides an overview of the Traversal Framework, a detailed description of the Neo4j Traversal Framework Java API and how to use it.
+--
+
+The Neo4j Traversal Framework Java API is a callback-based, lazily-executed way of specifying desired movements through a graph in Java.
+Some traversal examples can be found on xref:java-embedded/traversal.adoc[].
+
+Although the Traversal Framework is less readable and more complex than the xref:5.0@cypher-manual:ROOT:index.adoc[Cypher query language], it
+offers a powerful approach to traversing the graph. The ability to dynamically make custom choices at each step of the traversal makes the process more
+expressive and potentially more performant than Cypher.
+
+[WARNING]
+====
+It is generally recommended to use Cypher wherever it is possible.
+
+However, when using the Traversal Framework keep in mind:
+
+* Cypher uses memory tracking which allows a query to be aborted if it occupies too much memory.
+However, when mixing Cypher with Java in the Traversal API (e.g. on a function), you may run out of memory.
+* Do not reuse objects fetched during a Traversal in another transaction. Instead, use their IDs to fetch new ones.
+====
+
+== When to use the Traversal Framework over Cypher
+* The Traversal Framework allows the use of any desired Java library to help in the evaluation of the traversal.
+* The Traversal Framework allows customized pruning during the traversal of a path. This could potentially improve the performance of a traversal.
+See <<traversal-java-api-evaluator, Evaluator>> for more information.
+* With Cypher, it is not possible to specify the order in which paths are expanded (e.g. depth-first).
+However, with the Traversal Framework it is possible to specify the <<traversal-java-api-branchselector,order of paths traversed>>.
```

Comment:
It's missing a space after the comma in `<<traversal-java-api-branchselector,order of paths traversed>>`, but it might work anyway?